chat_auth/models.py

```diff
@@ -0,0 +1,139 @@
+from django.db import models
+from django.contrib.auth.models import (
+    BaseUserManager, AbstractBaseUser
+)
+
+
+class UserManager(BaseUserManager):
+    """
+    This is the manager that is being used in the Custom User model.
+    """
+
+    def create_user(self, email, username, password=None):
+        """
+        Creates and saves a User with the given email and password.
+        """
+        if not email:
+            raise ValueError('Users must have an email address')
+        if not username:
+            raise ValueError('Users must have a username')
+
+        if self.model.objects.filter(email=email):
+            raise ValueError('Email is taken already.')
+
+        if self.model.objects.filter(username=username):
+            raise ValueError('Username is taken already.')
+
+        user = self.model(
+            email=self.normalize_email(email),
+        )
+
+        user.username = username
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_staffuser(self, email, username, password):
+        """
+        Creates and saves a staff user with the given email and password.
+        """
+        user = self.create_user(
+            email,
+            username=username,
+            password=password,
+        )
+        user.staff = True
+        user.save(using=self._db)
+        return user
+
+    def create_superuser(self, email, username, password):
+        """
+        Creates and saves a superuser with the given email and password.
+        """
+        user = self.create_user(
+            email,
+            username=username,
+            password=password,
+        )
+        user.staff = True
+        user.admin = True
+        user.save(using=self._db)
+        return user
+
+
+class User(AbstractBaseUser):
```

Comment:
We will be modifying this model to have more functionalities in future.
Such as profile picture